http-server/http-server.go

```diff
@@ -3,34 +3,72 @@ package httpserver
 import (
 	"bytes"
 	"context"
+	"crypto/subtle"
+	"crypto/x509"
 	"encoding/json"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
 	"net/http"
+	"os"
 	"runtime"
 	"time"
 
+	grpcRuntime "github.com/grpc-ecosystem/grpc-gateway/runtime"
 	"github.com/packethost/pkg/log"
 	"github.com/pkg/errors"
 	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"github.com/tinkerbell/tink/protos/hardware"
+	"github.com/tinkerbell/tink/protos/template"
+	"github.com/tinkerbell/tink/protos/workflow"
 )
 
 var (
 	gitRev         = "unknown"
 	gitRevJSON     []byte
+	grpcListenAddr = "localhost:42113"
 	httpListenAddr = ":42114"
+	authUsername   = os.Getenv("TINK_AUTH_USERNAME")
+	authPassword   = os.Getenv("TINK_AUTH_PASSWORD")
 	startTime      = time.Now()
 	logger         log.Logger
 )
 
 // SetupHTTP setup and return an HTTP server
 func SetupHTTP(ctx context.Context, lg log.Logger, certPEM []byte, modTime time.Time, errCh chan<- error) {
 	logger = lg
+
+	cp := x509.NewCertPool()
+	ok := cp.AppendCertsFromPEM(certPEM)
+	if !ok {
+		logger.Error(errors.New("parse cert"))
```

Comment:
please add returned error code to logging

Reply:
What are you referring to with "returned error code"? If you meant the `ok`, that is just a boolean

Reply:
agree